src/core.rs

```diff
@@ -1046,6 +1100,16 @@ impl Core {
         }
     }
 
+    fn retry_bootstrap(&mut self) {
+        self.crust_service.stop_bootstrap();
+        self.state = State::Disconnected;
+        for &connection in self.proxy_map.keys() {
+            self.crust_service.drop_node(connection);
+        }
+        self.proxy_map.clear();
+        self.crust_service.bootstrap(0u32, Some(CRUST_DEFAULT_BEACON_PORT));
```

Comment:
Shouldn't this be `self.crust_service.bootstrap_with_blacklist` so we don't try and immediately bootstrap off the same peer?

Also, I'm not sure if we maybe need to wait for a small time before retrying?  Possibly we need to see how hard on the single beacon listener immediate rebootstrapping is.  If it can handle it, we're fine.  I'm just imagining 20 peers all repeatedly clamouring to bootstrap flooding the beacon listener.  If it *is* a problem, I think Crust should handle it itself (i.e. Crust should add a delay between bootstrap attempts if required), but until then we can get the example working by adding a sleep here temporarily.  Just a thought :)